shell/platform/android/test/io/flutter/embedding/android/FlutterActivityTest.java

```diff
@@ -38,6 +38,21 @@ public void itCreatesDefaultIntentWithExpectedDefaults() {
     assertEquals(FlutterView.TransparencyMode.opaque, flutterActivity.getTransparencyMode());
   }
 
+  @Test
+  public void itDestroysNewEngineWhenIntentIsMissingParameter() {
+    // All clients should use the static members of FlutterActivity to construct an
+    // Intent. Missing extras is an error. However, Flutter has number of tests that
+    // don't seem to use the static members of FlutterActivity to construct the
+    // launching Intent, so this test explicitly verifies that even illegal Intents
+    // result in the automatic destruction of a non-cached FlutterEngine, which prevents
+    // the breakage of memory usage benchmark tests.
+    Intent intent = new Intent(RuntimeEnvironment.application, FlutterActivity.class);
+    ActivityController<FlutterActivity> activityController = Robolectric.buildActivity(FlutterActivity.class, intent);
+    FlutterActivity flutterActivity = activityController.get();
+
+    assertTrue(flutterActivity.shouldDestroyEngineWithHost());
+  }
+
```

Comment:
Have more tests where a mockito engine is passed into the activity/fragment via the cache or by subclassing and provideFlutterEngine and check that flutterEngine.destroy is not called?

Reply:
Testing any lifecycle methods is outside what we seem to be able to do with our current testing infrastructure. Maybe we could assemble some e2e tests for that purpose, but even those seem to take a while to create. I did add some testing logic to cover the use-case you mentioned.

Reply:
Robolectric in general can test lifecycle events. We should have the infra for this. But maybe the problem is something other than just the raw ability for the test harness to drive lifecycle events?